src/LICM.cpp

```diff
@@ -316,6 +316,13 @@ class GroupLoopInvariants : public IRMutator2 {
     }
 
     Expr reassociate_summation(const Expr &e) {
+        if (e.type().is_float()) {
+            // Don't reassociate float exprs; it doesn't play well with -ffast-math.
+            // TODO: It might be safe to do this if 'strict_float' is enabled;
+            // consider revisiting this after that feature lands.
+            return e;
```

Comment:
I think you want to return IRMutator2::mutate(e)

Or change the Sub and Add visitor below to do this for floats.

Otherwise integer expressions buried inside float expressions (e.g. the index of a float load) won't get visited.

Reply:
Done